src/components/DishCard/DishCard.tsx

```diff
@@ -18,61 +21,49 @@ interface DishCardProps {
   dishTitleStyling?: React.CSSProperties;
   currencyIconSize?: React.CSSProperties;
   priceTextContainerStyling?: React.CSSProperties;
+  dishContentContainerStyling?: React.CSSProperties;
 }
 
 const DishCard: FC<DishCardProps> = (props) => {
+  const { updateDish, setIsModalActive } = useModalContext();
+  const isMobile = useIsMobile();
+  const isTablet = useIsTablet();
+  const isMobileOrTable = isMobile || isTablet;
+  const handleModalClick = () => {
+    setIsModalActive(true);
+    updateDish(props.dish);
+    if (isMobileOrTable) window.scrollTo(0, 0);
+  };
```

Comment:
Adding a comment here, or exporting this functionality into another function with a meaningful name would result in a cleaner code.

Reply:
@nitzanpap  done